src/main/java/com/revature/controllers/TripController.java

```diff
@@ -1,137 +1,219 @@
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP GET method (/trips/driver/{driveId})
+	 *
+	 * @param driverId represents the driver's ID.
+	 * @return A list of trips matching the driver's ID.
+	 */
+	@ApiOperation(value = "Returns trips by driver ID", tags = {"Driver", "Trip"})
+	@GetMapping("/driver/{driverId}")
+	public ResponseEntity<List<TripDTO>> getTripsByDriverId(@PathVariable("driverId") int driverId) {
+		List<Trip> tripsRaw = tripService.getTripsByDriverId(driverId);
+		List<TripDTO> trips = new ArrayList<>();
+		for (Trip trip : tripsRaw) {
+			trips.add(new TripDTO(trip));
+		}
+
+		if (!trips.isEmpty()) return ResponseEntity.ok(trips);
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP GET method (/trips/rider/{riderId})
+	 *
+	 * @param riderId represents the rider's ID.
+	 * @return A list of trips matching the rider's ID.
+	 */
+	@ApiOperation(value = "Returns trips by rider ID", tags = {"Rider", "Trip"})
+	@GetMapping("/rider/{riderId}")
+	public ResponseEntity<List<TripDTO>> getTripsByRiderId(@PathVariable("riderId") int riderId) {
+		List<Trip> tripsRaw = tripService.getTripsByRiderId(riderId);
+		List<TripDTO> trips = new ArrayList<>();
+		for (Trip trip : tripsRaw) {
+			trips.add(new TripDTO(trip));
+		}
+
+		if (!trips.isEmpty()) return ResponseEntity.ok(trips);
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP POST method (/trips)
+	 * @param trip represents the new Trip object being sent.
+	 * @return The newly created object along with a 201 code.
+	 */
+	@ApiOperation(value = "Adds a new trip")
+	@PostMapping
+	public ResponseEntity<TripDTO> addTrip(@Valid @RequestBody Trip trip) {
+		Optional<Trip> newTrip = tripService.getTripById(trip.getTripId());
+
+		if (newTrip.isPresent()) return ResponseEntity.badRequest().build();
+
+		return ResponseEntity.status(201).body(new TripDTO(tripService.addTrip(trip)));
+	}
+
+	/**
+	 * HTTP PUT method (/trips)
+	 *
+	 * @param trip represent the updated Trip object being sent.
+	 * @return The newly updated Trip object.
+	 */
+	@ApiOperation(value = "Updates a trip by ID", tags = {"Trip"})
+	@PutMapping
+	public ResponseEntity<TripDTO> updateTrip(@Valid @RequestBody Trip trip) {
+		Optional<Trip> existingTrip = tripService.getTripById(trip.getTripId());
+
+		if (existingTrip.isPresent()) return ResponseEntity.status(201).body(new TripDTO(tripService.updateTrip(trip)));
+
+		return ResponseEntity.badRequest().build();
+	}
+	
+	/**
+	 * HTTP POST method (/trips/{tripId}/rider)
+	 * 
+	 * @param tripId represents the id of the trip the person is joining
+	 * @param rider represents the User joining the trip.
+	 * @return The newly updated Trip object.
+	 */
+	@ApiOperation(value = "Adds a user to a trip", tags = {"Rider", "Trip"})
+	@PostMapping("/{tripId}/{riderId}")
+	public ResponseEntity<TripDTO> updateTripRider(@PathVariable("tripId") int tripId, @PathVariable("riderId") int riderId){
```

Comment:
We want @ QueryParam instead of @ PathVariable. QueryParams are the information after the ? in the url. Then the endpoint in the comment above needs to be updated to match.

Reply:
How do you suggest preventing the end point collision this is going to cause with the existing trip delete method?

Reply:
`@PostMapping("/join")` and `@DeleteMapping("/leave")`?

Reply:
They have different methods, so they don't clash.

Reply:
They only clash if they have the same endpoint and the same method.

Reply:
They won't clash with each other, but they will clash with the default methods the other team made that currently create and destroy trips (both of which are also "/" end points with post and delete).

Reply:
They won't be just "/" I think. Wouldn't they be /trips/rider?